internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -266,6 +281,39 @@ func evaluateReadOnlyState() bool {
 	return false
 }
 
+func (cmi *ClusterManagerImpl) punchHeartBeat(rvList []dcache.RawVolume) {
+	hostname, err := os.Hostname()
+	common.Assert(err != nil, "Error getting hostname in punchHeartBeat  %v", err)
+	listMyRVs(rvList)
+	hbData := dcache.HeartbeatData{
+		IPAddr:        rvList[0].IPAddress,
+		NodeID:        cmi.nodeId,
+		Hostname:      hostname,
+		LastHeartbeat: uint64(time.Now().Unix()),
+		RVList:        rvList,
+	}
+
+	// Marshal the data into JSON
+	data, err := json.MarshalIndent(hbData, "", "  ")
+	//Adding Assert because error capturing can just log the error and continue because it's a schedule method
+	common.Assert(err != nil, "Error marshalling heartbeat data %+v : error - %v", hbData, err)
+
+	// Create and update heartbeat file in storage with <nodeId>.hb
+	err = mm.UpdateHeartbeat(cmi.nodeId, data)
+	common.Assert(err != nil, "Error updating heartbeat file with nodeId %s in storage: %v", cmi.nodeId, err)
+	log.Trace("AddHeartBeat: Heartbeat file updated successfully")
+}
+
+func listMyRVs(rvList []dcache.RawVolume) {
+	for index, rv := range rvList {
+		_, availableSpace, err := common.GetDiskSpaceMetricsFromStatfs(rv.LocalCachePath)
+		common.Assert(err != nil, "Error getting disk space metrics for path %s for punching heartbeat: %v", rv.LocalCachePath, err)
```

Comment:
sorry the Assert cannot take a printf like string (I remember I misled you).
It does take variable parameters and logs all, if you want a formatted string, pls use sprintf and then pass that string to assert as a single param.

and this error, though unlikely, can happen, so we should also handle the error. Assert is good fro debug builds to catch if it happens, but in prod if it happens we need to handle it.
A good way to handle would be to set available space to 0, if we cannot query it.